### ISSUE TITLE
Postpone installing Impala and Dart EMR clusters until provisioning script is finished

### ISSUE THREAD

**Issue by maybeiambatman** [RetailMeNotSandbox/dart#144]
Launching EMR clusters on AWS is failing about 1/3rd of the time. This is what our AWS reps have told us:

> Since the issue is occurring intermittently, there is no definite way to reproduce it from my side in order to determine the root cause of it. I was able to confirm from your failed clusters that the launch is failing because EMR's node provisioning script (which uses Puppet to configure and install Hadoop and other applications) is getting failed.
> This mostly happens when a custom bootstrap action script introduces library conflicts with the EMR node provisioning script. As there are no logs relating to EMR provision-node script on the failed nodes, it is difficult to tell why the script failed in the first place.
> However, since we know for sure that this issue is happening because your bootstrap action script is in some way interfering with the EMR provision-node script, I would like to suggest you to modify your bootstrap action script (one which runs in the background) to wait until the provision-node script has finished successfully. Once the provision-node is finished, your script can execute its logic. This will ensure that your script does not interfere with EMR's node configuration process.
> In order to check whether the provision-node script has finished successfully or not, you can parse the /emr/instance-controller/lib/info/job-flow-state.txt file and look for "localInstance" block. In that block, you will see a "nodeProvisionCheckinRecord" block with a "status" key with "SUCCESSFUL" as its value.
